src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.SemanticChangeProcessor.cs

```diff
@@ -48,6 +48,20 @@ private sealed class SemanticChangeProcessor : IdleProcessor
                     _pendingWork = new Dictionary<DocumentId, Data>();
 
                     Start();
+
+                    AsyncProcessorTask.ContinueWith(
+                        _ =>
+                        {
+                            foreach (var (documentId, data) in _pendingWork)
+                            {
+                                data.AsyncToken.Dispose();
+                            }
+
+                            _pendingWork.Clear();
+                        },
+                        CancellationToken.None,
+                        TaskContinuationOptions.ExecuteSynchronously,
+                        TaskScheduler.Default);
```

Comment:
Would benefit from comments explaining what's going on here.  A comment also explaining that this should run unilaterally, regardless of how AsyncProcessorTask completes might be good.